### PR TITLE
Add blackbody PBR node implementations

### PR DESCRIPTION
**Pull request by nicolassavva-autodesk**
Address the missing targets for the PBR blackbody node implementation ( #1356 )
using the following approximation: [Wikipedia: Planckian Locus Approximation](https://en.wikipedia.org/wiki/Planckian_locus#Approximation)
Add a simple Standard Surface material example using the above blackbody node for emission.
Reenable blackbody node tests.

### CODE REVIEW

**Comment by jstone-lucasfilm** [comment]
This looks really promising, thanks @nicolassavva-autodesk.  Let us know if you run into any issues with the Autodesk CLA, and the Linux Foundation is usually very responsive if you file a support ticket using the link above.

**Comment by nicolassavva-autodesk** [comment]
Thanks Jonathan
I just re-authorized the Autodesk CLA with the account in question

File: libraries/pbrlib/genglsl/mx_blackbody.glsl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{
+    float xc, yc;
+    float t, t2, t3, xc2, xc3;
+
+    t = 1000.0 / temperatureKelvin;
+    t2 = t * t;
+    t3 = t * t * t;
+
+    // if value outside valid range of approximation clamp to accepted temperature range
+    clamp(temperatureKelvin, 1667.0, 25000.0);
```
Comment (niklasharrysson) [inline libraries/pbrlib/genglsl/mx_blackbody.glsl:16]: Should you not assign the result here? Like so:

`temperatureKelvin = clamp(temperatureKelvin, 1667.0, 25000.0);`

File: libraries/pbrlib/genosl/mx_blackbody.osl [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+{
+    float xc, yc;
+    float t, t2, t3, xc2, xc3;
+
+    t = 1000.0 / temperature;
+    t2 = t * t;
+    t3 = t * t * t;
+
+    // if value outside valid range of approximation clamp to accepted temperature range
+    clamp(temperature, 1667.0, 25000.0);
```
Comment (niklasharrysson) [inline libraries/pbrlib/genosl/mx_blackbody.osl:11]: Same here with assigning the result.

**Comment by niklasharrysson** [comment]
Looks great @nicolassavva-autodesk , I just noted an issue with the clamps above.

**Comment by nicolassavva-autodesk** [comment]
Good catch Niklas.
Added a quick fix for the missing clamping

**Review by jstone-lucasfilm — COMMENTED** [review]
This looks great, thanks @nicolassavva-autodesk.  My one recommendation would be to move the example material from the Standard Surface examples folder to the test suite, where it would be a better fit:
https://github.com/AcademySoftwareFoundation/MaterialX/tree/main/resources/Materials/TestSuite/pbrlib/bsdf

The example material could additionally be simplified to a simple nodegraph containing the `blackbody` node, rather than making it a Standard Surface material with pure emission.

**Review by jstone-lucasfilm — APPROVED** [review]
This looks good to me, thanks @nicolassavva-autodesk!

**Comment by krohmerNV** [comment]
is there an MDL implementation missing now?

**Comment by lgritz** [comment]
I hope I'm not opening a can of worms on a topic that has long since been settled, but is this function merely trying to return the color corresponding to a blackbody temperature, normalized to some standard luminance? Or is it supposed to be returning the actual emittance of the blackbody (W/m^2 or whatever)?

**Pull request merged**